impl/src/main/java/eu/darkbot/impl/galaxy/GalaxyInfoImpl.java

```diff
@@ -0,0 +1,132 @@
+package eu.darkbot.impl.galaxy;
+
+import eu.darkbot.api.game.galaxy.GalaxyGate;
+import eu.darkbot.api.game.galaxy.GateInfo;
+import eu.darkbot.util.XmlUtils;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class GalaxyInfoImpl implements eu.darkbot.api.game.galaxy.GalaxyInfo {
+    private final SpinResultImpl spinResult = new SpinResultImpl(this);
+
+    private final Map<GalaxyGate, GateInfo> gates = new HashMap<>();
+    private final Map<SpinResultImpl.ItemType, SpinResultImpl.SpinInfoImpl> itemCache = new HashMap<>();
```

Comment:
```suggestion
```
Get rid of this cache stuff, no longer needed if you keep the spin result impl and the results inside it